app/routes/owner/repositories.js

```diff
@@ -1,31 +1,52 @@
 import TravisRoute from 'travis/routes/basic';
 import config from 'travis/config/environment';
 import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
 
 export default TravisRoute.extend({
   features: service(),
+  insights: service(),
 
   needsAuth: false,
 
   queryParams: {
     page: {
       refreshModel: true
     },
+    tab: {
+      refreshModel: true
+    },
   },
 
-  model({ page }, transition) {
-    const limit = config.pagination.profileReposPerPage;
-    const offset = (page - 1) * limit;
-    const owner = this.paramsFor('owner').owner;
-    const type = 'byOwner';
-    const sort_by = 'default_branch.last_build:desc'; // eslint-disable-line
+  model({ page, tab }, transition) {
+    if (tab === 'insights') {
+      // TODO: Refactor owner route to use ember-data. Apparently owner is coming from an ajax call in the owner route.
+      let owner = { ...this.modelFor('owner') };
+      owner.isUser = owner['@type'] === 'user';
```

Comment:
Cool, that's different than what I initially tried 😄 but unfortunately it didn't seem to work. Maybe something about some of the adapters/serializers would need to be adjusted? I got it to kind of work, but it was a bit convoluted:

```
      const model = this.modelFor('owner');
      const pushModel = {
        type: model['@type'], // 'owner'
        id: model.id,
        attributes: {
          ...model,
        }
      };
      const owner = this.store.push({ data: pushModel });
```

And when I tried this in the owner route it broke a few things. I'm willing to give it more of a shot, and maybe pair on it, but for now I'd rather prioritize some other things.

Reply:
Yeah, sure, I thought it might just work out of the box (as long as we are currently loading users and organizations, which have the same payload 🤔). Did you try `this.store.pushPayload(model['@type'], model)`?

If it requires additional serialization tweaking, then it's probably outside of this PR's scope. In this case your solution perfectly works 👍

Reply:
Cool 😄 Yeah it's weird, it seems like it should work out of the box. I tried a bunch of stuff, but I only tried `pushPayload` a few times, and I don't remember if I tried that specifically. So I'll give that a quick try now :) Thanks!

Reply:
Unfortunately that didn't end up working either. So yeah I think for this PR I'll leave the current solution as is.